pkgs/halo2-eth-membership/src/eth_membership.rs

```diff
@@ -675,6 +648,107 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_eff_ecdsa_verification() -> Result<()> {
+        let path = "configs/eth_membership.config";
+        let circuit_params: CircuitConfig = serde_json::from_reader(
+            File::open(path)
+                .map_err(|e| anyhow!(e))
+                .with_context(|| format!("The circuit config file does not exist: {}", path))?,
+        )
+        .map_err(|e| anyhow!(e))
+        .with_context(|| format!("Failed to read the circuit config file: {}", path))?;
+
+        let (ecdsa_inputs, test_inputs) = mock_eff_ecdsa_input(PRIV_KEY)?;
+
+        let merkle_proof = mock_merkle_proof(&test_inputs.address)?;
+
+        let eth_membership_inputs = EthMembershipInputs::<
+            secp256k1::Fp,
+            secp256k1::Fq,
+            Secp256k1Affine,
+        >::new(ecdsa_inputs, merkle_proof);
+
+        let mut halo2_wasm = Halo2Wasm::new();
+
+        halo2_wasm.config(circuit_params);
+
+        let mut circuit =
+            EthMembershipCircuit::<secp256k1::Fp, secp256k1::Fq, Secp256k1Affine>::new(
+                &halo2_wasm,
+                eth_membership_inputs,
+            )?;
+
+        circuit.verify_membership();
+
+        let params = ParamsKZG::<E>::setup(K, OsRng);
+
+        // Load params
+        halo2_wasm.load_params(&serialize_params_to_bytes(&params));
+
+        // Generate VK
+        halo2_wasm.gen_vk();
+
+        // Generate PK
+        halo2_wasm.gen_pk();
+
+        // Get the public instance inputs
+        let instances = halo2_wasm.get_instance_values_ext(INSTANCE_COL)?;
+
+        let start = Instant::now();
+
+        // Verify Eff ECDSA
+        println!("Verifying Eff ECDSA Proof");
+
+        let is_eff_ecdsa_valid = verify_efficient_ecdsa(
+            test_inputs.msg_hash,
+            test_inputs.r,
+            test_inputs.is_y_odd,
+            &instances,
+        )?;
+
+        println!("- Is Eff ECDSA valid? {}", is_eff_ecdsa_valid);
+
+        assert!(is_eff_ecdsa_valid, "Eff ECDSA is not valid");
+
+        let duration = start.elapsed();
+        let duration_in_minutes = duration.as_secs_f64() / 60.0;
+        println!("Test executed in: {:.2?} seconds", duration);
+        println!("Test executed in: {:.2?} minutes", duration_in_minutes);
+
```

Comment:
```suggestion
```

we don't need this either
the test runner already reports test execution time
![image](https://github.com/user-attachments/assets/29ac1f4a-c0ef-4cf2-9de7-ded53a3381a8)


even if if your goal is to have a more accurate time for benchmarking purposes, then a test file is not the right place to do it

Reply:
Technically a benchmark would be measured by a test has loops through many different configs for the K value as you can see in this test: https://github.com/RiverRuby/halo2-lib-wasm/blob/9e22a0452add207800a4aa585e345cc9fb50a549/halo2-ecc/src/secp256k1/tests/ecdsa.rs#L63

But here that was also in the early dev steps where I wanted to get a sense how long it takes for each step.